crates/store/re_types/definitions/rerun/blueprint/archetypes/dataframe_query.fbs

```diff
@@ -49,3 +50,38 @@ table DataframeQuery (
     time_range_queries: rerun.blueprint.components.TimeRangeQueries ("attr.rerun.component_optional", nullable,order: 500);
 }
 
+
+
+/// The query for the dataframe view.
+table DataframeQueryV2 (
+    "attr.rerun.scope": "blueprint"
+) {
+    // --- Optional ---
+
+    /// The timeline for this query.
+    ///
+    /// If unset, the time panel's timeline is used and stored.
+    timeline: rerun.blueprint.components.TimelineName ("attr.rerun.component_optional", nullable, order: 100);
+
+    /// If set, a range filter is applied.
+    ///
+    /// Note: will be unset as soon as `timeline` is changed.
+    //TODO(ab): should this use the list-with-timelines trick?
+    range_filter: rerun.blueprint.components.RangeFilter ("attr.rerun.component_optional", nullable, order: 200);
+
+    /// Whether the filter by event feature is active.
+    // Note: we use an explicit field here such that the filter column is not lost when the filter is deactivated.
+    filter_by_event_active: rerun.blueprint.components.FilterByEventActive ("attr.rerun.component_optional", nullable, order: 300);
+
+    /// The column used when the filter by event feature is used.
+    ///
+    /// Note: only valid if the entity/component exists on `timeline`.
+    filter_by_event_column: rerun.blueprint.components.ComponentColumnSelector ("attr.rerun.component_optional", nullable, order: 400);
+
+    /// Should empty cells be filled with latest-at queries?
+    apply_latest_at: rerun.blueprint.components.ApplyLatestAt ("attr.rerun.component_optional", nullable, order: 500);
+
+    /// Selected columns. If unset, all columns are selected.
+    selected_columns: rerun.blueprint.components.SelectedColumns ("attr.rerun.component_optional", nullable, order: 600);
```

Comment:
Unless we change the `select` terminology in the data API, I strongly suggest to keep the `selected` part here as well as in the code (next PR, mainly). That is, even if we decide to opt for "visibility" on the UI side (which is still up for debate).